client.go

```diff
@@ -480,3 +484,82 @@ ExitForLoop:
 		}
 	}
 }
+
+func runHTTPLatencyTest(test *ethrTest) {
+	uri := test.session.remoteAddr
+	ui.printMsg("hi")
+	ui.printMsg("uri=%s", uri)
+	uri = "http://" + uri + ":" + httpLatencyPort
+
+	buff := make([]byte, test.testParam.BufferSize)
+	for i := uint32(0); i < test.testParam.BufferSize; i++ {
+		buff[i] = 'x'
+	}
+	
+	rttCount := test.testParam.RttCount
+	latencyNumbers := make([]time.Duration, rttCount)
+	ui.printMsg("%d",rttCount)
+	tr := &http.Transport{DisableCompression: true}
+	client := &http.Client{Transport: tr}
+ExitForLoop:
+	for {
+	ExitSelect:
+		select {
+		case <-test.done:
+			break ExitForLoop
+		default:
+			for i := uint32(0); i < rttCount; i++ {
+				s1 := time.Now()
+				response, err := client.Post(uri, "text/plain", bytes.NewBuffer(buff))
+				if err != nil {
+					break ExitSelect
+				} else {
+					if response.StatusCode != http.StatusOK {
+						break ExitSelect
+					}
+					contents, err := ioutil.ReadAll(response.Body)
+					response.Body.Close()
+					if err != nil {
+						break ExitSelect	
+					}
+					ethrUnused(contents)
+				}
+				
+				e2 := time.Since(s1)
+				latencyNumbers[i] = e2
+			}
+			// TODO temp code, fix it better, this is to allow server to do
+			// server side latency measurements as well.
+			sum := int64(0)
+			for _, d := range latencyNumbers {
+				sum += d.Nanoseconds()
+			}
+			elapsed := time.Duration(sum / int64(rttCount))
+			sort.SliceStable(latencyNumbers, func(i, j int) bool {
+				return latencyNumbers[i] < latencyNumbers[j]
+			})
+			//
```

Comment:
Can we move this code to common function? A function that takes latencyNumbers and rttCount as input and returns avg, min, max, p50 etc.

Reply:
Done